game-app/game-core/src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -1744,7 +1744,7 @@ private void showHistory() {
     inGame.set(false);
     setWidgetActivation();
     final GameData clonedGameData;
-    try (GameData.Unlocker ignored = data.acquireReadLock()) {
+    try (GameData.Unlocker ignored = data.acquireWriteLock()) {
```

Comment:
Why the write lock here all of the sudden?

Reply:
We're removing a listener on the data, which is a write operation.